apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -242,96 +240,128 @@ impl ApplyTo for NamedSelection {
     }
 }
 
-// $typenames is a special variable for referring to literal typenames. See
-// note in selection_set.rs for more detail.
-pub(super) const TYPENAMES: &str = "$typenames";
```

Comment:
Does this mean we're injecting typenames a different way? `->echo`, maybe?

Currently `$typenames` are exposed to users in debug info, so whatever we replace them with will be also. Something to keep in mind since we don't want to tell users that methods exist until we've added a couple of validations, I think. (cc @lennyburdette)

Reply:
Why do we need to expose anything related to `$typenames` or `$->echo` in debug info?

Reply:
I’m not concerned with hiding the typename implementation, we don’t have enough users. If someone finds it and tries something weird, good on em